client/my-sites/plans-v2/plans-header.tsx

```diff
@@ -53,13 +53,9 @@ const PlansHeader = ( { context }: { context: PageJS.Context } ) => {
 			?.map( ( { productSlug } ) => productSlug )
 			.filter( ( productSlug ) => JETPACK_PRODUCTS_LIST.includes( productSlug ) ) ?? [];
 
-	// When coming from in-connect flow, the url contains 'source=jetpack-plans' query param.
-	const isInConnectFlow = useMemo(
-		() => context.query?.source && context.query.source === 'jetpack-plans',
-		[ siteId ]
-	);
+	// When coming from in-connect flow, the url contains 'source=jetpack-connect-plans' query param.
+	const isInConnectFlow = context.query?.source === 'jetpack-connect-plans';
 
-	// TODO: Maybe make Notice dismissal persistent?
```

Comment:
We won't need to persist a dismissal since this banner will only be shown after at the end of the in-place connection flow and not when users click the Plans link in wp-admin. In other words, the banner will appear very infrequently.